index.js

```diff
@@ -23,50 +23,77 @@ function getMetadata (opts, dir, cb) {
   var props = []
   if (!opts.name) props.push(['productName', 'name'])
   if (!opts['app-version']) props.push('version')
-  if (!opts.version) props.push(['dependencies.electron', 'devDependencies.electron'])
+  if (!opts.version) {
+    props.push([
+      'dependencies.electron',
+      'devDependencies.electron',
+      'dependencies.electron-prebuilt',
+      'devDependencies.electron-prebuilt'
+    ])
+  }
 
   // Name and version provided, no need to infer
   if (props.length === 0) return cb(null)
 
   // Search package.json files to infer name and version from
-  getPackageInfo(props, dir, function (err, result) {
-    if (err) {
-      // `get-package-info` exploded looking for `electron`. Try `electron-prebuilt` instead
-      props.pop()
-      props.push(['dependencies.electron-prebuilt', 'devDependencies.electron-prebuilt'])
-      getPackageInfo(props, dir, function (err, result) {
-        if (err) return cb(err)
-        return inferNameAndVersionFromInstalled('electron-prebuilt', opts, result, cb)
+  getPackageInfo(props, dir, (err, result) => {
+    if (err && err.missingProps) {
+      var requiredProps = ['productName', 'dependencies.electron']
+      var missingProps = err.missingProps.map(prop => {
+        return Array.isArray(prop) ? prop[0] : prop
       })
-    } else {
-      return inferNameAndVersionFromInstalled('electron', opts, result, cb)
+
+      // Callback w/ error if there are required props that are missing
+      if (missingProps.filter(prop => requiredProps.find(reqProp => prop === reqProp)).length !== 0) {
+        var messages = missingProps.map(function (missingProp) {
+          missingProp = Array.isArray(missingProp) ? missingProp[0] : missingProp
+          if (missingProp === 'productName') {
+            return 'Unable to determine application name. Please specify an application name\n\n' +
+              'For more information, please see\n' +
+              'https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#name\n'
+          }
+
+          if (missingProp === 'dependencies.electron') {
+            return 'Unable to determine Electron version. Please specify an Electron version\n\n' +
+              'For more information, please see\n' +
+              'https://github.com/electron-userland/electron-packager/blob/master/docs/api.md#version\n'
+          }
+        })
+
+        err.message = messages.join('\n') + '\n' + err.message
```

Comment:
May want to hide original error message, since the message we add should be descriptive enough without it?